linc_convert/utils/nifti_header.py

```diff
@@ -0,0 +1,243 @@
+from linc_convert.utils.unit import to_nifti_unit
+from linc_convert.utils.zarr_config import NiftiConfig
+
+
+def _try_open_zarr_array(path: Path) -> "ZarrArray":  # noqa: F821
+    """
+    Attempt to open a Zarr array at `path`.
+
+    Returns the opened array on success, or raises the last exception.
+    """
+    last_err: Exception | None = None
+    for v in (2, 3):
+        try:
+            return open_array(path, mode="r", zarr_version=v)
+        except Exception as e:  # noqa: BLE001 - we want to keep original exception
+            # types
+            last_err = e
+    # If we get here, all attempts failed.
+    assert last_err is not None
+    raise last_err
+
+
+def load_nifti_header_from_file(path: str | Path) -> "NiftiHeader":
+    """
+    Load a NIfTI header from a file path.
+
+    Supports:
+      - Standard NIfTI files: `.nii`, `.nii.gz`
+      - Zarr containing Nifti Header Array: `.zarr`
+      - A Zarr Array containing a NIfTI header
+
+    Parameters
+    ----------
+    path : str | pathlib.Path
+        Path to a NIfTI file or Zarr directory.
+
+    Returns
+    -------
+    NiftiHeader
+        A nibabel NIfTI header object.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the path does not exist.
+    ValueError
+        If the path exists but is not a recognized NIfTI file or Zarr directory.
+    Exception
+        If Zarr opening fails for all supported versions.
+    """
+    p = Path(path)
+    if not p.exists():
+        raise FileNotFoundError(f"Path does not exist: {p}")
+
+    suffix = "".join(p.suffixes).lower()
+
+    # Standard NIfTI files
+    if suffix.endswith(".nii") or suffix.endswith(".nii.gz"):
+        img = nib.load(str(p))
+        return img.header.copy()
+
+    if suffix.endswith(".zarr") or p.is_dir():
+        zarr_target = p
+        candidate = p / "nifti"
+        if candidate.exists():
+            zarr_target = candidate
+        return load_nifti_header_from_zarray(zarr_target)
+
+    warnings.warn(
+        f"Cannot load NIfTI header from {p!r}: not a recognized NIfTI file or Zarr "
+        f"directory.",
+        stacklevel=2,
+    )
+    raise ValueError(f"Unrecognized NIfTI header source: {p!r}")
+
+
+def load_nifti_header_from_zarray(path: str | Path) -> "NiftiHeader":
+    """
+    Load a NIfTI header from a Zarr array storing the raw NIfTI header bytes.
+
+    This function attempts Zarr format v2 first, then v3 (or vice versa if you
+    change the order in `_try_open_zarr_array`). It reads the header bytes once,
+    infers the correct nibabel header class via `get_nibabel_klass`, and
+    constructs the header with `from_fileobj(check=False)` (since the buffer
+    contains exactly the header bytes, not a full image).
+
+    Parameters
+    ----------
+    path : str | pathlib.Path
+        Path to the Zarr array (directory) containing NIfTI header bytes, or a
+        directory that contains a `nifti` Zarr array.
```

Comment:
I might be mistaken but it looks like this function does not look within a directory for the `/nifti` Zarr array.

```suggestion
        Path to the Zarr array (directory) containing the NIfTI header bytes.
```